cellprofiler/__main__.py

```diff
@@ -1,4 +1,14 @@
 from __future__ import print_function
+
+import os
+import os.path
+import sys
+if sys.platform.startswith("win") and hasattr(sys, 'frozen'):
+    # Allow us to package java with windows builds
+    if 'JAVA_HOME' not in os.environ:
```

Comment:
This _could_ be the following, which might have some better clarity:
```
# Set JAVA_HOME to be CP_JAVA_HOME if it is available, otherwise set to an empty string
# (os.environ does not accept None as a valid environment variable)
os.environ['JAVA_HOME'] = os.environ.get('JAVA_HOME') or os.environ.get('CP_JAVA_HOME') or ''
```

Reply:
I don't think we want to do the empty string setting because I don't think we want to actually set JAVA_HOME in case we need Java to do its normal thing to find copies of itself.  Just for maximum safety.

Reply:
Okay, then I'd just have this logic be `if 'JAVA_HOME' not in os.environ and 'CP_JAVA_HOME` in os.environ: <assigment>`

Reply:
I can add more clarity to the comment though.

Reply:
inside the sys.platform.startswith, or outside it?

Reply:
Oh sorry, no your current implementation looks good!